### PR TITLE
Visual element screen/UI

### PR DESCRIPTION
**Pull request by abhianand102**
## Description

- A versatile React Native component designed to render either an `Image` or a `Lottie` animation based on the provided source

- The component dynamically adjusts its rendering logic and `style properties` to accommodate different types of visual content, offering enhanced flexibility for UI development

- Visual Element Component Implementation:- 
   - Created the Visual Element component which accepts an image or a Lottie animation as its source
  
   - Type Definitions:
     - Defined Lottie Source type to handle different formats of Lottie animation sources
     - Defined Visual Element Props type for the component props, enhancing type safety and code clarity

  - Styling:
     - Utilized the styles object from ./styles to apply consistent styling to the component
     - Ensured that styles can be dynamically adjusted based on the optional props provided

### CODE REVIEW

File: src/FRONTEND/stylescomponent.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { StyleSheet } from "react-native";
```
Comment (Aashirvaad-P) [inline src/FRONTEND/stylescomponent.tsx]: There is no FRONTEND folder in src as per "develop" branch.
1. Make sure to follow the proper repo structure, not dump files anywhere.
2. Do follow the proper conventions of files and folders.

File: @types/lottie-react-native.d.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+declare module 'lottie-react-native' {
```
Comment (Aashirvaad-P) [inline @types/lottie-react-native.d.ts]: this seems to be coming from node_modules.
Nothing from node_modules should be pushed in the repo.

File: Liveliness/LivelinessScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { View, Text, Button, TouchableOpacity, StyleSheet, Image } from 'react-native';
+import { RNCamera } from 'react-native-camera';
+import styles from './styles';
+
+const LivelinessScreen = () => {
```
Comment (Jha2508) [inline Liveliness/LivelinessScreen.tsx]: documentation

File: Liveliness/Styles.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { StyleSheet } from "react-native";
+const styles = StyleSheet.create({
```
Comment (Jha2508) [inline Liveliness/Styles.tsx]: This can be a simple `.ts` file

File: Liveliness/Styles.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        color: '#333',
+        fontSize: 18,
+        marginBottom: 10,
+    },
+    dot: {
+        width: 15,
+        height: 15,
+        marginBottom:8,
+        marginRight:5,
+        //marginRight: 75,
```
Comment (Jha2508) [inline Liveliness/Styles.tsx]: remove comments if not needed

File: src/FRONTEND/visualElement.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+type LottieSource = string | { uri: string } | object;
+
+const VisualElement: React.FC<{
+  imageSource: ImageSourcePropType;      // Image source
+  lottieSource: LottieSource;            // Lottie Source
+  width?: number;                        // defining width
+  height?: number;                       // defining height
+  cornerRadius?: number;                 // definig cornerRadius
+  rotationAngle?: number;                // definibg rotationAngle
+}> = ({
```
Comment (Jha2508) [inline src/FRONTEND/visualElement.tsx]: Make a type of `VisualElementProps` and pass it here

File: src/FRONTEND/visualElement.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        },
+      ]}
+    >
+      <Image source={imageSource} style={styles.image} />
+      <LottieView source={lottieSource} autoPlay loop style={styles.lottie} />
+    </Animated.View>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (Jha2508) [inline src/FRONTEND/visualElement.tsx]: move styles to different file

**Comment by Aashirvaad-P** [comment]
Make the changes in file naming. Would check code after that. Also resolve the conflicts.

**Comment by Aashirvaad-P** [comment]
This is a component and not a screen. Make the changes accordingly.

File: src/assets/Animation - 1719487285162.json [hunk 8]
```diff
@@ -0,0 +1 @@
...3,0.333],"y":[0,0]},"t":7.548,"s":[256,135]},{"i":{"x":[0.667,0.667],"y":[1,1]},"o":{"x":[0.167,0.167],"y":[0,0]},"t":14.259,"s":[215,310]},{"i":{"x":[0.833,0.833],"y":[1,1]},"o":{"x":[0.333,0.333],"y":[0,0]},"t":26,"s":[205,205]},{"i":{"x":[0.667,0.667],"y":[1,1]},"o":{"x":[0.167,0.167],"y":[0,0]},"t":30.193,"s":[256,160]},{"i":{"x":[0.667,0.667],"y":[1,1]},"o":{"x":[0.167,0.167],"y":[0,0]},"t":42.774,"s":[210,260]},{"t":47.806640625,"s":[205,205]}],"ix":2},"p":{"a":0,"k":[0,0],"ix":3},"r":{"a":0,"k":321,"ix":4},"nm":"Rectangle Path 1","mn":"ADBE Vector Shape - Rect","hd":false},{"ty":"st","c":{"a":0,"k":[0.305882352941,0.305882352941,0.886274569642,1],"ix":3},"o":{"a":0,"k":100,"ix":4},"w":{"a":0,"k":80,"ix":5},"lc":2,"lj":1,"ml":4,"bm":0,"nm":"Stroke 1","mn":"ADBE Vector Graphic - Stroke","hd":false},{"ty":"tr","p":{"a":0,"k":[-127.949,376.736],"ix":2},"a":{"a":0,"k":[0.042,102.405],"ix":1},"s":{"a":0,"k":[102.5,102.487],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Rectangle 1","np":3,"cix":2,"bm":0,"ix":1,"mn":"ADBE Vector Group","hd":false}],"ip":0,"op":144,"st":0,"bm":0}],"markers":[]}
```
Comment (Aashirvaad-P) [inline src/assets/Animation - 1719487285162.json]: Rename the file properly.

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**